neuraxle/checkpoints.py

```diff
@@ -19,185 +19,442 @@
         """
-        Set checkpoint Path
+        Returns the checkpoint file path for a data input id
 
-        :param path: checkpoint path
+        :param checkpoint_path:
+        :param current_id:
+        :return:
         """
         raise NotImplementedError()
 
     @abstractmethod
-    def read_checkpoint(self, data_container: DataContainer) -> DataContainer:
+    def save_current_id_checkpoint(self, checkpoint_path: str, current_id, data_input, expected_output) -> bool:
         """
-        Read checkpoint data to get the data inputs and expected output.
+        Save the given current id, data input, and expected output.
 
-        :param data_container: data inputs to save
-        :return: checkpoint data container
+        :param checkpoint_path:
+        :param current_id:
+        :param data_input:
+        :param expected_output:
+        :return:
         """
         raise NotImplementedError()
 
     @abstractmethod
-    def save_checkpoint(self, data_container: DataContainer) -> DataContainer:
+    def read_current_id_checkpoint(self, checkpoint_path: str, current_id) -> Tuple:
         """
-        Save checkpoint for data inputs and expected outputs so that it can
-        be loaded by the checkpoint pipeline runner on the next pipeline run
+        Read the data inputs, and expected outputs for the given current id.
 
-        :param data_container: data inputs to save
-        :return: saved data container
+        :param checkpoint_path:
+        :param current_id:
+        :return: tuple(current_id, checkpoint_data_input, checkpoint_expected_output)
+        :rtype: tuple(str, Iterable, Iterable)
         """
         raise NotImplementedError()
 
 
-class PickleCheckpointStep(BaseCheckpointStep):
+class PickleDataContainerSaver(BaseDataContainerSaver):
     """
-    Create pickles for a checkpoint step data inputs, and expected_outputs
-    to eventually be able to load them using the checkpoint pipeline runner.
+    PickleDataContainerSaver is used by DataContainerCheckpointer so save,
+    and load the checkpoint files for all of the current ids inside a data container.
     """
+    def __init__(self, execution_mode: ExecutionMode):
+        if execution_mode == ExecutionMode.FIT:
+            self.checkpoint_file_name_suffix = ExecutionMode.FIT.value
+
+        if execution_mode == ExecutionMode.TRANSFORM:
+            self.checkpoint_file_name_suffix = ExecutionMode.TRANSFORM.value
 
-    def __init__(self, cache_folder: str = DEFAULT_CACHE_FOLDER):
-        BaseCheckpointStep.__init__(self)
-        self.cache_folder = cache_folder
+        if execution_mode == ExecutionMode.FIT_TRANSFORM:
+            self.checkpoint_file_name_suffix = ExecutionMode.FIT_TRANSFORM.value
 
-    def read_checkpoint(self, data_container: DataContainer) -> DataContainer:
+    def save_current_id_checkpoint(self, checkpoint_path: str, current_id, data_input, expected_output) -> bool:
```

Comment:
What is saving an id checkpoint?

(Disclaimer: I'm dumb writing this, I somehow know but I want a better name)

Reply:
I cannot find a better name ? save_checkpoint_for_current_id( ??